### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Uthmanbello**
## Highlights
- Created a new Rails app called 'hello-rails-react'.
- Initialized the project with Git.
- Installed needed node dependencies.
- Created a babel.config.js in the root folder of the project.
- Updated webpack.config.js to include the babel loader.
- Made sure that the project has a Postgres database set up.
- Created a table for storing messages and created 5 different greetings.
- Created an API endpoint that selects a random greeting from the table.
- Created a static view that will be the root of the app.
- Created the App component with react-router and rendered it in the static view.
- Created the Greeting component that will display a greeting. Set it up as a route in the App component.
- Created a store, an action, and a reducer that will connect to the API endpoint to get the random greeting.
- Displayed the random greeting in the Greeting component.

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Uthmanbello 🤝 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not following the correct git flow, would you submit a PR that is coming from a feature branch into the base branch? Ex: `feature/initialize-project` -> `development`

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Uthmanbello** [comment]
@descholar-ceo , 
Fixed the issue.

Thanks a lot.

File: app/javascript/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  useEffect(() => {
+    dispatch(getRandomGreeting());
+  }, [dispatch]);
+
+  return (
+    <div>
+      <h4>Random Greeting</h4>
+      <p>{greeting}</p>
+    </div>
+  );
```
Comment (KDlamini) [inline app/javascript/Greeting.js:17]: Random greeting displayed successfully. Great job!!! 👏🏾💯 

image

**Review by KDlamini — APPROVED** [review]
### Hi @Uthmanbello 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements for setting up a 'Hello world' project as one app with Webpack. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞

✔️  Professional pull request
✔️  API endpoints implemented correctly
✔️  React client renders Greetings component using the route in App component
✔️  Well documented README

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Uthmanbello** [comment]
@KDlamini ,

Thanks a lot

**Pull request merged**